examples/asr/triton-inference-server/asr_ctc/model_repo/asr_ctc/1/model.py

```diff
@@ -129,16 +125,34 @@ def initialize(self, args):
           * model_name: Model name
         """
         model_config = json.loads(args["model_config"])
-        model_name = model_config["parameters"]["model_name"]["string_value"]
+        parameters = model_config["parameters"]
+        model_name_or_path = parameters["model_name_or_path"]["string_value"]
 
         assert args["model_instance_kind"] == "GPU"
 
-        self.dtype = torch.bfloat16
+        dtype_map = {
+            "bfloat16": torch.bfloat16,
+            "float16": torch.float16,
+            "float32": torch.float32
+        }
+        dtype = dtype_map[parameters["dtype"]["string_value"]]
+        self.dtype = dtype
+        if model_name_or_path.endswith(".nemo"):
+          asr_model = nemo_asr.models.ASRModel.restore_from(model_name_or_path)
+        else:
+          asr_model = nemo_asr.models.ASRModel.from_pretrained(model_name_or_path)
+        ctc_decoding = CTCDecodingConfig()
+        ctc_decoding.strategy = parameters['decoding_strategy']['string_value']
+        if ctc_decoding.strategy == 'beam_search':
+            ctc_decoding.beam_size = int(parameters['beam_size']['string_value'])
+        asr_model.change_decoding_strategy(ctc_decoding)
+        self.model = asr_model.cuda()
+        self.model.encoder.freeze()
+        self.model.decoder.freeze()
+        self.model.eval()
+        self.model.to()
```

Comment:
Just to check, did you run this code yet? I'm not sure if `model.to()` with no argument is valid.

Reply:
yes, i ran it. deleted this line now.